MagneticField/GeomBuilder/src/BaseVolumeHandle.h

```diff
@@ -0,0 +1,219 @@
+#ifndef MagneticField_GeomBuilder_BaseVolumeHandle_H
+#define MagneticField_GeomBuilder_BaseVolumeHandle_H
```

Comment:
I am uncomfortable with this proposal. The original class is part of the implementation and not meant to be exposed to the package public interface. Note that all the corresponding functionality was formerly included in the regular lib of the package (not in the plugin); and this is now broken by moving the implementation of the "new version" to plugins/dd4hep while keeping the old one in the regular lib. I would either keep things as they are and move back the new version to src once we drop the old DDD, or move all corresponding parts of the old implementation to plugins.

Reply:
The original `MagneticField/GeomBuilder/plugins/VolumeBasedMagneticFieldESProducer.cc` has included for a long time `MagneticField/GeomBuilder/src/MagGeoBuilderFromDDD.h`. In the same way, the new `MagneticField/GeomBuilder/plugins/dd4hep/DD4hep_VolumeBasedMagneticFieldESProducer.cc` includes `MagneticField/GeomBuilder/src/DD4hep_MagGeoBuilder.h`. `BaseVolumeHandle.h` is no longer included by name by any file in the `plugins` or `plugins/dd4hep` directories.

I am leaving this overall file organization question to Nicola and will simply follow the existing pattern for now.

Reply:
I agree with Carl that this is the best solution now. The (unfortunate) initial splitting of code between src and plugins is historical since in early days the package was designed to support linking the MF to standalone code outside CMSSW (which is no longer the case since ages). When we will drop the old DDD it will be necessary to reshuffle files to build a single plugin anyhow so it will be a good time for re-thinking the file organization.